crates/telio-relay/src/derp.rs

```diff
@@ -726,8 +734,9 @@ impl Runtime for State {
                     },
                     // On tick send derp poll request to derp stream
                     Some((permit, _)) = wait_for_tx(&c.comms_direct.tx, poll_timer_tick) => {
-                        if config.enable_polling {
+                        if config.enable_polling || config.server_keepalives.poll_keepalive.is_some() {
```

Comment:
I think that maybe, for simplification of things, it makes sense to leave `enable_polling` responsible just for polling(as it is currently on mainline) but the new flag just control which endpoint on DERP it connects to and controls server->client logic instead?

This way the same parameter for polling could be reused when connecting to derp/v1 and derp/v2 and the flags would be separated, resulting in this simple understanding of behaviour:
- `enable_polling` - enables polling optimisation
- `poll_keepalive` - connects to derp/v2 turning off derp->libtelio keepalives(this has no effect on libtelio)
- `enable_polling` + `poll_keepalive` -> enable polling optimisation and connect to derp/v2 to disable derp->libtelio optimisation


This should make both flags separated and they would work separately and together.
This would result in simpler logic. For example now `poll_keepalives` take preference over `enable_polling` which is hidden from the API user.

What's your opinion on this?

Reply:
Changed as we discussed